frontend/app/src/main/java/com/example/speechbuddy/viewmodel/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,131 @@
+    }
+
+    fun clearPassword() {
+        passwordCheckInput = ""
+        passwordInput = ""
+    }
+
+    private fun validatePassword() {
+        if (isValidPassword(passwordInput)) {
+            _uiState.update { currentState ->
+                currentState.copy(
+                    isValidPassword = true,
+                    error = null
+                )
+            }
+        }
+    }
+
+    private fun validatePasswordCheck() {
+        if (passwordInput == passwordCheckInput) {
+            _uiState.update { currentState ->
+                currentState.copy(
+                    error = null
+                )
+            }
+        }
+    }
+
+    fun resetPassword(navController: NavHostController) {
+        if (!isValidPassword(passwordInput)) { // Check password length
+            _uiState.update { currentState ->
+                currentState.copy(
+                    isValidPassword = false,
+                    error = ResetPasswordError(
+                        type = ResetPasswordErrorType.PASSWORD,
+                        messageId = R.string.false_new_password
+                    )
+                )
+            }
+        } else if (passwordInput != passwordCheckInput) { // Check password equality
+            _uiState.update { currentState ->
+                currentState.copy(
+                    error = ResetPasswordError(
+                        type = ResetPasswordErrorType.PASSWORD_CHECK,
+                        messageId = R.string.false_new_password_check
+                    )
+                )
+            }
+        } else {
+            viewModelScope.launch {
+                var accessToken = MainApplication.token_prefs.getAccessToken()
+                accessToken = "Bearer $accessToken"
+
+                repository.resetPassword(
+                    accessToken = accessToken,
+                    AuthResetPasswordRequest(
+                        password = passwordInput
+                    )
+                ).collect { result ->
+                    Log.d("test", result.toString())
+                    when (result.status) {
+                        Status.SUCCESS -> {
+                            navController.navigate("login")
+                        }
+
+                        Status.ERROR -> {
+                            _uiState.update { currentState ->
+                                currentState.copy(
+                                    isValidPassword = false,
+                                    error = ResetPasswordError(
+                                        type = ResetPasswordErrorType.PASSWORD_CHECK,
+                                        messageId = R.string.resset_password_error
+                                    )
+                                )
+                            }
+                        }
+
+                        else -> {}
+                    }
+                }
+            }
+        }
+        clearPassword()
+    }
+
+}
```

Comment:
```suggestion
                }
                clearInputs()
            }
        }
    }
}
```

지금은 에러가 나도 사용자가 쓴 인풋이 자동으로 지워져서 서버로 요청이 갈 때에만 지워지는 것으로 바꿨습니다